AstSemantics.md

```diff
@@ -1,14 +1,22 @@
 # Abstract Syntax Tree Semantics
 
-WebAssembly code is represented as an Abstract Syntax Tree (AST) where each node
-represents an expression. Each function body consists of a list of expressions.
-All expressions and operators are typed, with no implicit conversions or overloading rules.
+WebAssembly code is represented in a form that can be interpreted either as an
+Abstract Syntax Tree (AST) where each node represents an expression, or as a
+typed stack machine, where each node represents an instruction. This document
+describes WebAssembly's semantics from an AST-oriented perspective, however
+WebAssembly can be naturally described from a stack machine perspective as well.
+
+Implementations need not build an actual AST or maintain an actual stack; they
+need only behave [as if](https://en.wikipedia.org/wiki/As-if_rule) they did so.
 
```

Comment:
fwiw I don't think describing wasm as a 'stack machine' will help the reader because: the nodes are still expressions and need not represent one value on the stack; and there are some in-order operators that do not fit the 'stack machine' 'instruction' description well. Perhaps after some of the changes under discussion it will fit the 'stack machine' description better, such as removing the `invalid`/`empty` type, and defining multiple value calls to push multiple values on a values stack, but there are a lot of issues to consider.